src/orion/algo/tpe.py

```diff
@@ -478,9 +537,16 @@ def _sample_real_point(self, dimension, below_points, above_points, is_log=False
 
         return new_point
 
-    def _sample_int_point(self, dimension, below_points, above_points):
+    def _sample_int_point(
+        self,
+        dimension: Integer,
+        below_points: numpy.ndarray | Sequence[numpy.ndarray],
+        above_points: numpy.ndarray | Sequence[numpy.ndarray],
+    ):
         """Sample one value for integer dimension based on the observed good and bad points"""
         low, high = dimension.interval()
+        assert isinstance(low, int)
```

Comment:
Another assert

Reply:
This assert is fine. Again, this is making the `TypeError` that you'd get anyway with the `range` below more explicit.

Reply:
This isn't for the user. It's for the devs.Since we already check for the Integer dim, there's no point in checking again, right?
This assert is the result of loose typing on the Dimension objects, really. If the annotation for `interval` on `Integer` gave `tuple[int, int]`, then I wouldn't mind taking out this assert.
Can we mark this as resolved?

Reply:
Yep, deferring discussion here: https://github.com/Epistimio/orion/issues/859